doc/advanced-topics.md

```diff
@@ -83,6 +83,50 @@ e.g.:
    }
 ```
 
+#### NGSI-LD data and metadata considerations
+
+When provisioning devices for an NGSI-LD Context Broker, `type` values should typically correspond to one of the
+following:
+
+-   `Property`, `Relationship`, `Geoproperty`
+-   Native JSON types (e.g. `String`, `Boolean`, `Float` , `Integer` `Number`)
+-   Temporal Properties (e.g. `Datetime`, `Date` , `Time`)
+-   GeoJSON types (e.g `Point`, `LineString`, `Polygon`, `MultiPoint`, `MultiLineString`, `MultiPolygon`)
+
+Most NGSI-LD attributes are sent to the Context Broker as _properties_. If a GeoJSON type or native JSON type is
+defined, the data will be converted to the appropriate type. Temporal properties should always be expressed in UTC,
+using ISO 8601. This ISO 8601 conversion is applied automatically for the `observedAt` _property-of-a-property_ metadata
+where present.
+
+Data for any attribute defined as a _relationship_ must be a valid URN.
+
+Note that when the `unitCode` metadata attribute is supplied in the provisioning data under NGSI-LD, the standard
+`unitCode` _property-of-a-property_ `String` attribute is created.
+
+Other unrecognised `type` attributes will be passed as NGSI-LD data using the following JSON-LD format:
+
+```json
+    "<property_name>": {
+       "type" : "Property",
+       "value": {
+          "@type":  "<property_type>",
+          "@value":  { string or object}
+      }
+    }
+```
+
+`null` values will be passed in the following format:
```

Comment:
Why did you choose to pass null values in the following format, is it a requirement ? I cannot find this mentionned in the ngsi-ld specification.

Reply:
This is a restriction based on the definitions of the [ESTI NGSI-LD specification](https://www.etsi.org/deliver/etsi_gs/CIM/001_099/009/01.02.01_60/gs_cim009v010201p.pdf)

> ### 5.2.1 Introduction
> Implementations shall support the data types defined by the clauses below. For each 
> member defined by each data type (including nested ones) a term shall be added to the 
> Core `@context`, as mandated by  clause 4.5. **None of the members described admit a 
> `null` value**, except when they are used in the context of an update operation (see
> clause 5.5.8) and implementations shall raise an error of type BadRequestData if a `null`
> value is encountered. 

and ...

> ### 5.5.8 JSON-LD Merge Patch Behaviour 
> 
> - For each member of the Fragment, whose value is `null`, contained by the target, the target member is removed. 

so basically, you cannot store `null` directly as `null` since this is the signal to delete an attribute. The work around is not to send a `null` directly, but to send a JSON-LD `object` which is equivalent to `null`.

Reply:
(late reply)

It's not clear to me, you say you cannot store ```null``` but you're still trying to store a  ```null``` value for the ```@value``` of ```Intangible```.
Do I miss something there ?

Reply:
What do you think about passing a default value? as you use that only for numbers, 0 seems to be a correct choice.

Reply:
> It's not clear to me, you say you cannot store `null` but you're still trying to store a `null` value 
> for the `@value` of Intangible. Do I miss something there ?

No that is precisely what I am attempting to do.   Directly sending `"value" : null` is not legal NGSI-LD hence the workaround of sending an _object equivalent to `null`_ -  `"value": { "@type":  "Intangible", "@value":  null }`

> What do you think about passing a default value? as you use that only for numbers, `0`
> seems to be a correct choice.

But `0` is a real number and may not be appropriate in all cases. To me,`null` better indicates *Undefined*.